src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -276,21 +293,23 @@
        :else
        [message/message message-data context keyboard-shown?])]))
 
-(defn scroll-handler
-  [event scroll-y animate-topbar-opacity? on-end-reached? animate-topbar-name?]
-  (let [content-size-y  (- (oops/oget event "nativeEvent.contentSize.height")
-                           (oops/oget event "nativeEvent.layoutMeasurement.height"))
-        current-y       (oops/oget event "nativeEvent.contentOffset.y")
-        scroll-distance (- content-size-y current-y)]
-    (when (and @on-end-reached? (pos? scroll-distance))
+(defn use-scroll-handler
+  [{:keys [scroll-y-animated-value
+           content-size-animated-value
+           animate-topbar-opacity?
+           on-end-reached?
+           animate-topbar-name?]}]
+  (fn [new-scroll-y new-content-size]
+    (when (and @on-end-reached? (pos? new-scroll-y))
       (reset! on-end-reached? false))
-    (if (< topbar-visible-scroll-y-value scroll-distance)
+    (if (< topbar-visible-scroll-y-value new-scroll-y)
```

Comment:
Hi, for reference related changes (more details checkout changes in https://github.com/status-im/status-mobile/pull/18055)